libs/small-steps-test/src/Control/State/Transition/Trace.hs

```diff
@@ -499,3 +504,55 @@ applySTSTest = applySTSOptsEither defaultOpts
           asoValidation = ValidateAll,
           asoEvents = EPDiscard
         }
+
+-- | We have pairs that look like [(s_n+1,dn), ... ,(s3,d2),(s2,d1)]
+--   We want pairs that look like [(sn,dn), ... ,(s2,d2),(s1,d1)]
```

Comment:
Maybe a bit of an explanation as to how this is helpful when working with traces?

Reply:
Added a lot of new comments, that explain the overall strategy.